src/app/modules/form/admin-form/admin-form.utils.ts

```diff
@@ -517,3 +547,41 @@ export const verifyUserBetaflag = (
         ),
       )
 }
+
+// Utility method to map an axios error to a GoGovError
+export const mapGoGovErrors = (error: AxiosError): GoGovError => {
+  type GoGovReturnedData = { message: string; type?: string }
+  // Hard coded error message from GoGov for URL Bad Request
+  // TODO: Verify with GoGov team if error response data can be different from general short link error
+  const urlFormatError = 'Only HTTPS URLs are allowed'
+
+  const responseData = error.response?.data as GoGovReturnedData
+
+  switch (error.response?.status) {
+    case StatusCodes.BAD_REQUEST:
+      // There can be three types of Bad Request from GoGov
+      // Short link already exists, which returns type=ShortUrlError
+      // Or validation error, which does not contain type
+      // Or if url is not https (like localhost), however, this should not happen as we prepend the app url in admin-form-controller
+      // TODO: Verify with GoGov team on response data shape
+      return responseData.type
+        ? new GoGovAlreadyExistError()
+        : responseData.message.includes(urlFormatError)
+        ? new GoGovServerError(
+            'GoGov server returned 400 for URL formatting error',
+          )
+        : new GoGovValidationError()
```

Comment:
nit: can this be in the same order as what you commented? either by changing this or the comments

Reply:
[a6c20ea](https://github.com/opengovsg/FormSG/pull/6714/commits/a6c20ea7794bf2452514eb34e2d2e822a81d84d4)